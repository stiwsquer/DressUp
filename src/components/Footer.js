import React from "react";

export default function Footer(props) {
  return (
    <footer>
      <div className="footer-wrapper">
        <h5>DressUp &copy;</h5>
        <ul>
          <li>
            <a href="#" title="twitter-social-media">
              {/* <img src="../icons/twitter.svg" alt="twitter-social-media" /> */}
            </a>
          </li>
          <li>
            <a href="#" title="youtube-social-media">
              {/* <img src="../icons/youtube.svg" alt="youtube-social-media" /> */}
            </a>
          </li>
          <li>
            <a href="#" title="instagram-social-media">
              <img
                src={require("../icons/heart.png")}
                alt="instagram-social-media"
              />
            </a>
          </li>
        </ul>
      </div>
    </footer>
  );
}
