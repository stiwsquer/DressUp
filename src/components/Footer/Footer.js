import React from "react";
import NavItem from "../NavItem/NavItem";
import "./style.scss";

export default function Footer() {
  return (
    <footer>
      <div className="footer-wrapper">
        <h1>
          <NavItem text="DressUp" linkTo="#" /> &copy;
        </h1>
        <ul>
          <li>
            <NavItem iconClass="fab fa-twitter social-media" linkTo="#" />
          </li>
          <li>
            <NavItem iconClass="fab fa-instagram social-media" linkTo="#" />
          </li>
          <li>
            <NavItem iconClass="fab fa-youtube social-media" linkTo="#" />
          </li>
        </ul>
      </div>
    </footer>
  );
}
