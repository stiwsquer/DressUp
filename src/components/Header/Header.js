import React from "react";
import { useSearchContext } from "../../context/SearchContext";
import Navigation from "../Navigation/NavigationV2";
import "./style.scss";

export default React.memo(function Header() {
  const [input] = useSearchContext();

  return (
    <header className="main-head">
      <Navigation />
      <h1>{input}</h1>

      <section className="sections"></section>
    </header>
  );
});
