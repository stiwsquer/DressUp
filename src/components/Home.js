import React from "react";
import gallery1 from "../images/gallery-1.2.jpg";
import gallery2 from "../images/gallery-2.1.jpg";
import gallery3 from "../images/gallery-3.1.jpg";
import home_intro from "../images/home-intro_1.jpg";

export default function Home(props) {
  return (
    <>
      <section className="home-intro"></section>
      <section className="home-description">
        <h2>
          Trendy Clothing & Accessories at Dress Up - An Online Dress Boutique
        </h2>
        <p>
          DRESS UP IS A WOMEN&apos;S CLOTHING BOUTIQUE CARRYING A VARIETY OF
          FITS, STYLES, AND PATTERNS FOR WOMEN OF ALL AGES, SHAPES, AND SIZES.
          WE OFFER A DIVERSE COLLECTION OF NEW PRODUCTS EVERY SINGLE WEEK,
          INCLUDING YOUR FAVORITE CLOTHING ITEMS: CUTE ROMPERS, MAXI DRESSES,
          KNIT SWEATERS, OVERSIZED TOPS, AND LIGHTWEIGHT CARDIGANS, AS WELL AS
          TRENDY SHOES, ACCESSORIES, JEWELRY, ACTIVEWEAR, LOUNGEWEAR, AND SO
          MUCH MORE. DRESS UP&apos;S TRENDY, YET WEARABLE CLOTHES WILL HAVE YOU
          FEELING LIKE YOUR BEST SELF, NO MATTER WHERE YOU&apos;RE HEADED. YOU
          CAN SHOP WITH US ONLINE OR AT ANY OF OUR RETAIL LOCATIONS ACROSS THE
          SOUTHEAST, INCLUDING STORES IN GEORGIA, ALABAMA, SOUTH CAROLINA, AND
          TENNESSEE. GET FREE SHIPPING ON ALL ORDERS OVER $49, AND FEEL FREE TO
          REACH OUT TO US AT 678.989.2006 WITH ANY QUESTIONS. DON&apos;T MISS
          OUT - COME SEE WHY DRESS UP IS ONE OF THE FASTEST GROWING ONLINE DRESS
          BOUTIQUES IN THE UNITED STATES!{" "}
        </p>
        <button>SHOP OUR NEWEST ARRIVALS</button>
      </section>
      <section className="home-gallery">
        <div className="gallery-new-shoes">
          <h2>NEW SHOES</h2>
          <p>
            We just dropped new fall shoe styles, and they are all ON SALE! Shop
            new combat boots, booties, mules, sneakers & more!
          </p>
          <button>SHOP SHOES</button>
        </div>
        <div className="gallery-accessories">
          <h2>Accessories</h2>
          <p>
            Take your fall outfit to the next level with new accessories! The
            perfect trendy handbag, dainty gold necklace, coin bracelet or
            statement earrings will upgrade any look.
          </p>
          <button>SHOP ACCESSORIES</button>
        </div>
        <img className="gallery1" src={gallery1} />
        <img className="gallery2" src={gallery2} />
        <img className="gallery3" src={gallery3} />
      </section>
    </>
  );
}
