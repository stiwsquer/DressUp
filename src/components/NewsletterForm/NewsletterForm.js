import React from "react";
import Button from "../Button/Button";
import Input from "../Input/Input";
import "./style.scss";

export default function NewsletterForm(props) {
  return (
    <form className="newsletter-form">
      <h4>Subscribe to our newsletter</h4>
      <p>Get the latest updates on new products and upcoming sales</p>
      <div>
        <Input
          isRequired={true}
          inputType="email"
          placeholder="Your email adress"
          inputId="email"
        />

        <Button linkTo="#">SUBSCRIBE</Button>
      </div>
    </form>
  );
}
