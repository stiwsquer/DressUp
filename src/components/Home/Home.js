import React from "react";
// import "./style.scss";
import { HomeIntro } from "./Home.style";

import HomeDescription from "../HomeDescription/HomeDescription";
import HomeGallery from "../HomeGallery/HomeGallery";
import HomeProducts from "../HomeProducts/HomeProducts";

export default React.memo(function Home() {
  return (
    <>
      <HomeIntro />
      <HomeDescription />
      <HomeGallery />
      <HomeProducts />
    </>
  );
});
