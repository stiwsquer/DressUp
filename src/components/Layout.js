import { func } from "prop-types";
import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Header from "./Header";
import Footer from "./Footer";
import Home from "./Home";
import NewsletterForm from "./NewsletterForm";
import SignIn from "./SignIn";
import Register from "./Register";

export default function Layout(props) {
  return (
    <Router>
      <Header />
      <main>
        <Switch>
          <Route path="/signIn">
            <SignIn />
          </Route>
          <Route path="/register">
            <Register />
          </Route>
          <Route path="/">
            <Home />
          </Route>
        </Switch>
      </main>
      <NewsletterForm />
      <Footer />
    </Router>
  );
}
