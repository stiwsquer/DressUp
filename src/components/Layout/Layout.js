import { func } from "prop-types";
import React from "react";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Header from "../Header/Header";
import Footer from "../Footer/Footer";
import Home from "../Home/Home";
import NewsletterForm from "../NewsletterForm/NewsletterForm";
import SignIn from "../SignIn/SignIn";
import Register from "../Register/Register";

export default function Layout() {
  return (
    <Router>
      <Header />
      <main>
        <Switch>
          <Route path="/signIn">
            <SignIn />
          </Route>
          <Route path="/register">
            <Register />
          </Route>
          <Route exact path="/">
            <Home />
          </Route>
        </Switch>
      </main>
      <NewsletterForm />
      <Footer />
    </Router>
  );
}
