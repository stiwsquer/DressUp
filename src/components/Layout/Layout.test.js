import Footer from "../Footer/Footer";
import Header from "../Header/Header";
import Home from "../Home/Home";
import NewsletterForm from "../NewsletterForm/NewsletterForm";
import Register from "../Register/Register";
import SignIn from "../SignIn/SignIn";
import Layout from "./Layout";

it("renders Layout component without crashing", () => {
  shallow(<Layout />);
});

it("contains one Switch component", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper.find("Switch")).toHaveLength(1);
});

it("contains three Route component", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper.find("Route")).toHaveLength(3);
});

it("contains NewsletterForm component", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper.contains(<NewsletterForm />)).toBe(true);
});

it("contains Footer component", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper.contains(<Footer />)).toBe(true);
});

it("contains one Header component", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper.contains(<Header />)).toBe(true);
});

it("contains one SignIn component", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper.contains(<SignIn />)).toBe(true);
});

it("contains Register component", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper.contains(<Register />)).toBe(true);
});

it("contains Home component", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper.contains(<Home />)).toBe(true);
});

it("Layour snapshot - shallow", () => {
  const wrapper = shallow(<Layout />);
  expect(wrapper).toMatchSnapshot();
});
