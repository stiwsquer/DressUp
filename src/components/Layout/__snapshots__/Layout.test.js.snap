// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layour snapshot - shallow 1`] = `
<BrowserRouter>
  <Memo(Header) />
  <main>
    <Switch>
      <Route
        path="/signIn"
      >
        <Memo(SignIn) />
      </Route>
      <Route
        path="/register"
      >
        <Memo(Register) />
      </Route>
      <Route
        exact={true}
        path="/"
      >
        <Memo(Home) />
      </Route>
    </Switch>
  </main>
  <Memo(NewsletterForm) />
  <Memo(Footer) />
</BrowserRouter>
`;
